gap/examples.gi

```diff
@@ -370,3 +370,889 @@ GeneralisedPetersenGraphCons);
+  D := Digraph(IsMutableDigraph, []);
+  DigraphAddVertices(D, Length(vertices));
+  for i in [1 .. Length(vertices) - 1] do
+    for j in [i + 1 .. Length(vertices)] do
+      bList := List([1 .. Length(vertices[i])],
+        x -> vertices[i][x] <> vertices[j][x]);
+      pos := Positions(bList, true);
+      if bList[1] then
+        if (SizeBlist(bList) = 2 and
+            vertices[j][pos[2]] = vertices[i][pos[1]] and
+            vertices[j][pos[1]] = vertices[i][pos[2]]) or
+            (SizeBlist(bList) = 1 and (not vertices[j][1] in vertices[i])) then
+          DigraphAddEdge(D, i, j);
+          DigraphAddEdge(D, j, i);
+        fi;
+      fi;
+    od;
+  od;
+  return D;
+end);
+
+InstallMethod(PermutationStarGraphCons,
+"for IsImmutableDigraph, integer, integer",
+[IsImmutableDigraph, IsInt, IsInt],
+function(filt, n, k)
+  local D;
+  D := MakeImmutable(PermutationStarGraphCons(IsMutableDigraph, n, k));
+  SetIsMultiDigraph(D, false);
+  SetIsSymmetricDigraph(D, true);
+  return D;
+end);
+
+InstallMethod(PermutationStarGraph, "for two integers", [IsInt, IsInt],
+{n, k} -> PermutationStarGraphCons(IsImmutableDigraph, n, k));
+
+InstallMethod(PermutationStarGraph, "for a function and two integers",
+[IsFunction, IsInt, IsInt], PermutationStarGraphCons);
+
+InstallMethod(WindmillGraphCons, "for IsMutableDigraph and two integers",
+[IsMutableDigraph, IsInt, IsInt],
+function(filt, n, m)
+  local D, i, K, nrVert;
+  if (not IsInt(m)) or (not IsInt(n)) or m < 2 or n < 2 then
+    ErrorNoReturn("the arguments <n> and <m> must be integers greater than 1,");
+  fi;
+  D := Digraph(IsMutableDigraph, []);
+  K := CompleteDigraph(n - 1);
+  nrVert := 1 + DigraphNrVertices(K) * m;
+  DigraphAddVertices(D, nrVert);
+  for i in [0 .. (m - 1)] do
+    DigraphAddEdges(D, DigraphEdges(K) + (i * DigraphNrVertices(K)));
+  od;
+  for i in [1 .. (DigraphNrVertices(D) - 1)] do
+    DigraphAddEdge(D, i, nrVert);
+    DigraphAddEdge(D, nrVert, i);
+  od;
+  return D;
+end);
+
+InstallMethod(WindmillGraphCons,
+"for IsImmutableDigraph, integer, integer",
+[IsImmutableDigraph, IsInt, IsInt],
+function(filt, n, m)
+  local D;
+  D := MakeImmutable(WindmillGraphCons(IsMutableDigraph, n, m));
+  SetIsMultiDigraph(D, false);
+  SetIsSymmetricDigraph(D, true);
+  return D;
+end);
+
+InstallMethod(WindmillGraph, "for two integers", [IsInt, IsInt],
+{n, m} -> WindmillGraphCons(IsImmutableDigraph, n, m));
+
+InstallMethod(WindmillGraph, "for a function and two integers",
+[IsFunction, IsInt, IsInt], WindmillGraphCons);
+
+InstallMethod(PathGraphCons, "for IsMutableDigraph and an integer",
+[IsMutableDigraph, IsInt],
+function(filt, n)
+  local D;
+  if (not IsInt(n)) or n < 1 then
+    ErrorNoReturn("the argument <n> must be an integer greater than 0,");
+  fi;
+  D := Digraph(IsMutableDigraph, []);
+  DigraphAddVertices(D, n);
+  DigraphAddEdges(D, List([1 .. (n - 1)], x -> [x, x + 1]));
+  return DigraphSymmetricClosure(D);
```

Comment:
You could just `return DigraphSymmetricClosure(ChainDigraph(IsMutableDigraph, n));`

Reply:
Agreed! In any case, even if `ChainDigraph` didn't exist, it's not worth it to create an empty digraph, add the necessary number of vertices, and then add the edges all in separate steps - you could just directly do:
```
D := Digraph(IsMutableDigraph, List([1 .. (n - 1)], x -> [x, x + 1]));
```